From 8e69b663d6ddef132041a1186f081fdd74d4a31d Mon Sep 17 00:00:00 2001
From: rofl0r <retnyg@gmx.net>
Date: Mon, 20 Jan 2014 21:31:34 +0100
Subject: [PATCH 3/3] remove inclusion of sysinfo.h in kernel.h

the declaration of struct sysinfo clashes with userspace.
it's not quite clear why that header was included from kernel.h,
as none of its functionality is needed.

Note: Modified for support backport to linux_libc_headers_3.18.131, to
match the linux_quic_3.18.131. This is for building kernel 3.18.131
with Yocto 2.5, which has a preferred libc header of 4.15.x

---
Signed-off-by: Khem Raj <raj.khem@gmail.com>
Modified-by: Steven MacDonald <smacdonald@sierrawireless.com>
Upstream-Status: Submitted

 include/uapi/linux/kernel.h | 2 ++
 1 file changed, 2 insertions(+)

Index: linux-4.8-rc4/include/uapi/linux/kernel.h
===================================================================
--- linux-4.8-rc4.orig/include/uapi/linux/kernel.h
+++ linux-4.8-rc4/include/uapi/linux/kernel.h
@@ -1,7 +1,9 @@
 #ifndef _UAPI_LINUX_KERNEL_H
 #define _UAPI_LINUX_KERNEL_H
 
+#ifdef __GLIBC__
 #include <linux/sysinfo.h>
+#endif
 
 /*
  * 'kernel.h' contains some often-used function prototypes etc
